test/outdated/outdated_test.dart

```diff
@@ -54,6 +54,13 @@ Future<void> main() async {
     expectMatchesGoldenFile(
         buffer.toString(), 'test/outdated/goldens/helptext.txt');
   });
+
+  test('no lockfile', () async {
+    await d.appDir({'foo': '^1.0.0'}).create();
+    await servePackages((builder) => builder..serve('foo', '1.2.3'));
```

Comment:
What if `2.0.0` exists, then the users should get a notice that constraint in `pubspec.yaml` can be bumped.

Reply:
Done